services/elasticache/pkg/resource/replication_group/custom_set_conditions.go

```diff
@@ -0,0 +1,60 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package replication_group
+
+import (
+	ackv1alpha1 "github.com/aws/aws-controllers-k8s/apis/core/v1alpha1"
+	svcapitypes "github.com/aws/aws-controllers-k8s/services/elasticache/apis/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// CustomUpdateConditions sets conditions (terminal) on supplied replication group
+// it examines supplied resource to determine conditions.
+func (rm *resourceManager) CustomUpdateConditions(
+	ko *svcapitypes.ReplicationGroup,
+	r *resource,
+	err error,
+) {
+	rgStatus := r.ko.Status.Status
+	if rgStatus != nil && *rgStatus != "create-failed" {
+		return
+	}
+
+	// Terminal condition
+	var terminalCondition *ackv1alpha1.Condition = nil
+	if ko.Status.Conditions == nil {
+		ko.Status.Conditions = []*ackv1alpha1.Condition{}
+	} else {
+		for _, condition := range ko.Status.Conditions {
+			if condition.Type == ackv1alpha1.ConditionTypeTerminal {
+				terminalCondition = condition
+				break
+			}
+		}
+		if terminalCondition != nil && terminalCondition.Status == corev1.ConditionTrue {
+			// some other exception already put the resource in terminal condition
+			return
+		}
+	}
+	if terminalCondition == nil {
+		terminalCondition = &ackv1alpha1.Condition{
+			Type: ackv1alpha1.ConditionTypeTerminal,
+		}
+		ko.Status.Conditions = append(ko.Status.Conditions, terminalCondition)
+	}
+
+	terminalCondition.Status = corev1.ConditionTrue
+	errorMessage := "Replication group status: create-failed"
```

Comment:
If all we're really needing a custom update condition method for is to set this string, then it might be easier to just change the `ExceptionsConfig.TerminalCodes` Go type from `[]string` to instead be a struct with some fields that can accomodate this custom error message. Perhaps even place this struct on the `OperationConfig` struct so you can vary the message by API call, too. For instance, above is hard-coding "create-failed" when you might want to have a message customized to the update failed path...

Reply:
This scenario is when a successful sdk method call response includes a property (ko.Status.Status), which help determine if the resource is in terminal state. Thus, needed to have it as a custom method.